cloudferrylib/utils/drivers/ssh_chunks.py

```diff
@@ -29,177 +25,178 @@
-                                             "Retrying... Attempt %s from %s.",
-                                             part, path_src, attempt, attempts_count)
-                                if attempt == attempts_count:
-                                    LOG.error("SSH chunks transfer of %s failed.",
-                                              path_src)
-                                    break
-                                    # TODO: save state:
-                                    # chunks count, volume info, and all metadata info
-                                    # with timestamp and reason, errors info, error
-                                    # codes
-                                continue
-
-                            #  Write chunk on destination
-                            if part == range(part_count)[0]:
-                                command = dd_dst_command % (dst_temp_dir,
-                                                            part,
-                                                            path_dst,
-                                                            0,
-                                                            part_size)
-                            elif part == range(part_count)[-1] and part_modulo:
-                                command = dd_dst_command % (dst_temp_dir,
-                                                            part,
-                                                            path_dst,
-                                                            part * part_size,
-                                                            part_modulo)
-                            else:
-                                command = dd_dst_command % (dst_temp_dir,
-                                                            part,
-                                                            path_dst,
-                                                            part * part_size,
-                                                            part_size)
-                            with hide('running'):
-                                # Because of password
-                                run(ssh_command %
-                                    (ssh_user_dst, host_dst,
-                                     'echo \'%s\' | sudo -S %s' % (ssh_sudo_pass_dst,
-                                                                   command)))
-
-                                LOG.info(
-                                    'Running: %s', ssh_command %
-                                    (ssh_user_dst, host_dst,
-                                     'echo \'%s\' | sudo -S %s' % ('<password>',
-                                                                   command)))
-
-    def _calculate_parts_count(self, data):
-        part_size = self.cfg.migrate.ssh_chunk_size
-
-        byte_size = data.get(
-            'byte_size',
-            utils.get_remote_file_size(data['host_src'],
```

Comment:
Looks like get_remote_file_size is not used anymore, can be deleted. It uses run directly and this is terrible. Many issues can be fixed by using runner.run.